pkg/apis/config/v1alpha1/resmgr/policy/balloons/config.go

```diff
@@ -220,11 +221,13 @@ type BalloonDef struct {
 	ShareIdleCpusInSame CPUTopologyLevel `json:"shareIdleCPUsInSame,omitempty"`
 	// PreferCloseToDevices: prefer creating new balloons of this
 	// type close to listed devices.
+	// +listType=set
 	PreferCloseToDevices []string `json:"preferCloseToDevices,omitempty"`
 	// PreferFarFromDevices: prefer creating new balloons of this
 	// type far from listed devices.
 	// TODO: PreferFarFromDevices is considered too untested for usage. Hence,
 	// for the time being we prevent its usage through CRDs.
+	// +listType=set
```

Comment:
I'm not sure if we should add validation to a field that is hidden. And here order matters, too.

Furthermore, as we fill PreferFarFromDevices automatically, I wonder what happens if the automation has added a device twice in the list, and the result would be stored into a yaml, then restored from the yaml, it would be quite surpising if validation would reject a yaml that seemed to be a working configuration.

I'd propose leaving this out.